src/PIL/IcoImagePlugin.py

```diff
@@ -52,6 +52,7 @@ def _save(im, fp, filename):
     sizes = list(sizes)
     fp.write(struct.pack("<H", len(sizes)))  # idCount(2)
     offset = fp.tell() + len(sizes) * 16
+    alt_images = {im.size: im for im in im.encoderinfo.get("append_images", [])}
     for size in sizes:
```

Comment:
It looks like if someone were to add an unusual sized image to append_images (e.g. (20,20)), it would not be saved unless it is added to the sizes parameter as well. I feel like this could be confusing behaviour. Perhaps adding `sizes.extend(alt_images.keys())` would be a good idea (sizes would have to be changed to a set to avoid duplicates).

Reply:
That is correct. I believe the ICNS saving currently suffers from the same issue. I don't know if it should be fixed for .ICOs; the documentation would need to mention the role of append_images either way, since if the behavior is changed as you suggest, it would then also save other sizes than the ones the user requested (via the sizes parameter).